src/ucs/type/status.h

```diff
@@ -42,9 +42,23 @@ typedef enum {
     UCS_ERR_TIMED_OUT              = -20,
     UCS_ERR_EXCEEDS_LIMIT          = -21,
     UCS_ERR_UNSUPPORTED            = -22,
+
```

Comment:
Some of the already existing errors could be considered "link errors" (TIMED_OUT, UNREACHABLE? ) 

ERR_SHMEM_SEGMENT: why does this error code specifically name an UCX client? Could it be "anonymized" ?